### ISSUE TITLE
vergy small loss while bad performance

### ISSUE THREAD

**Issue by Frankcht** [facebookresearch/denoiser#49]
I should say it is an excellent project firstly.

Once I cloned the codes and tried to train the sample dataset (debug), the training process has no problem. The validation loss is around 0.05 and the performance of the enhanced noisy file is good. Then I tried to replace the debug dataset by my dataset:
1. move my dataset to `dataset`
2. generate the clean and noisy .json files with `make_debug.sh`
3.  modify the `conf/config.yaml` and build a new `conf/dset/mydata.yaml` for my dataset
The new training process has no errors and the validation loss is around 0.0008. But when I check the enhanced performance of files in `output/ex/samples`, the level of the enhanced wav files is almost zero. The level of the input noisy files is around 500~1000, so there should be mistakes somewhere. But I dont know whether I missed something or my dataset has some problem?

Thanks in advance.

**Comment by adefossez** [comment]
What do you call the level? What is the relative scale of the noise and clean speech in your case?
You can try to add a `print` statement there https://github.com/facebookresearch/denoiser/blob/master/denoiser/enhance.py#L83 where normalization happens, which could explain what your output audio is almost silent (for instance if a single time step has a huge value).

```
    print(wav.std().item(), wav.abs().max().item())
    wav = wav / max(wav.abs().max().item(), 1)
```
and report the typical numbers you get.

**Comment by Frankcht** [comment]
> 
> 
> What do you call the level? What is the relative scale of the noise and clean speech in your case?
> You can try to add a `print` statement there https://github.com/facebookresearch/denoiser/blob/master/denoiser/enhance.py#L83 where normalization happens, which could explain what your output audio is almost silent (for instance if a single time step has a huge value).
> 
> ```
>     print(wav.std().item(), wav.abs().max().item())
>     wav = wav / max(wav.abs().max().item(), 1)
> ```
> 
> and report the typical numbers you get.

Hi. I tried to print the std and max values then returned:
std = 1e-3 ~ 4e-3
max = 6e-2 ~ 1e-1

They are the resonable values?
Thx!

**Comment by adefossez** [comment]
Those seem reasonable. See my earlier question then : What do you call the level? What is the relative scale of the noise and clean speech in your case?

Also what is the size of your custom dataset?

**Comment by Frankcht** [comment]
> 
> 
> Those seem reasonable. See my earlier question then : What do you call the level? What is the relative scale of the noise and clean speech in your case?
> 
> Also what is the size of your custom dataset?

Thx for your reply!

1. 'level' means the amplitude value when I open the wav files with Audition. It seems to be the sample value instead of db scale.
2. According to the sample values (level), clean speech is about twice higher than noise. That is 6db according to db scale.
3. My dataset is total 444 hours and it was split into 100000 wav pieces. Each piece is 16s. Specially, train data contains 60000 pieces (266 hours), validation data has 30000 pieces (133 hours) and test data has 10000 pieces (66 hours).

**Comment by adefossez** [comment]
I'm surprised you get such a small training loss in that case. 
Can you take a small subset of your train set and try with it?
did you change some other parameters in the config?
